tests/testsuite/test.rs

```diff
@@ -4792,6 +4792,21 @@ error: test failed, to rerun pass `--test t1`
 [RUNNING] tests/t2.rs (target/debug/deps/t2[..])
 error: test failed, to rerun pass `--test t2`
 
+Caused by:
+  process didn't exit successfully: `[ROOT]/foo/target/debug/deps/t2[..]` (exit [..]: 4)
+note: test was terminated via a signal. stderr might be truncated. use `--nocapture` flag to disable output buffering.
+",
+        )
+        .with_status(4)
+        .run();
+
+    p.cargo("test --test t2 -- --nocapture")
```

Comment:
Can you split these new test cases out into a commit (first in the PR).  It should pass, showing the behavior of `master`.  The commit with all of the other changes should then update the test to still pass.  This will make it easier to see (1) the tests work and (2) what behavior changed